extensions/in-memory/assetindex-memory/src/test/java/org/eclipse/dataspaceconnector/metadata/memory/InMemoryAssetIndexLoaderTest.java

```diff
@@ -0,0 +1,88 @@
+package org.eclipse.dataspaceconnector.metadata.memory;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.eclipse.dataspaceconnector.spi.asset.AssetIndexLoader;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.types.domain.asset.Asset;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataAddress;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+import java.util.Random;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.easymock.EasyMock.niceMock;
+
+// this test aims at testing the thread-safety of the asset index
+public class InMemoryAssetIndexLoaderTest {
+
+    private final Random random = new Random();
+    private AssetIndexLoader assetIndexLoader;
+
+    @Test
+    void insert() {
+        var asset = createAsset("test-asset", UUID.randomUUID().toString());
+        var dataAddress = createDataAddress(asset);
+        assetIndexLoader.insert(asset, dataAddress);
+
+        assertThat(((InMemoryAssetIndex) assetIndexLoader).getAssets()).hasSize(1);
+        assertThat(((InMemoryAssetIndex) assetIndexLoader).getDataAddresses()).hasSize(1);
+    }
+
+    @Test
+    void insert_illegalParams() {
+        assertThatThrownBy(() -> assetIndexLoader.insert(null, niceMock(DataAddress.class))).isInstanceOf(NullPointerException.class);
```

Comment:
Maybe should better avoid mocking data classes in favour of using dummy objects

Reply:
is there any particular reason for this?

Reply:
I think that mocks should be used to verify behavior and not state. this test is quite simple and here's not that big issue, but generally mocking data classes could bring to unmaintainable and valueless tests

Reply:
ok, I don't have a strong opinion about that. I'll change it.